GitHubTeams.ps1

```diff
@@ -153,8 +161,39 @@ filter Get-GitHubTeam
         'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
     }
 
-    return (Invoke-GHRestMethodMultipleResult @params |
-        Add-GitHubTeamAdditionalProperties)
+    $result = Invoke-GHRestMethodMultipleResult @params | Add-GitHubTeamAdditionalProperties
+
+    if ($PSBoundParameters.ContainsKey('TeamName'))
+    {
+        $team = $result | Where-Object -Property name -eq $TeamName
+
+        if ($null -eq $team)
+        {
+            $message = "Team '$TeamName' not found"
+            Write-Log -Message $message -Level Error
+            throw $message
+        }
+        else
+        {
+            $uriFragment = "/teams/$($team.id)"
```

Comment:
Bringing this [up again](https://github.com/microsoft/PowerShellForGitHub/pull/222#discussion_r446556438).  You had previously said that you didn't add this code, that you just changed its indent, but that's not what I'm seeing here.  This call to the `/teams/$($team.id)` endpoint appears to be new with this change, but it's also to [an endpoint that has been deprecated](https://developer.github.com/v3/teams/#get-a-team-legacy).  We definitely shouldn't be adding new code to deprecated endpoints.  You'd also likely want to update `New-*` and `Set-*` to be that type as well (and an update to `.OUTPUTS` to mention this alternate output type).